code/modules/power/geothermal/geothermal_extension.dm

```diff
@@ -0,0 +1,34 @@
+/datum/extension/geothermal_vent
+	base_type = /datum/extension/geothermal_vent
+	expected_type = /atom
+	flags = EXTENSION_FLAG_IMMEDIATE
+	var/pressure_min = 1000
+	var/pressure_max = 2000
+	var/steam_min = 30 SECONDS
+	var/steam_max = 1 MINUTE
+	var/tmp/next_steam = 0
+	var/datum/effect/effect/system/steam_spread/steam
+
+/datum/extension/geothermal_vent/New()
+	..()
+	START_PROCESSING(SSobj, src)
+
+/datum/extension/geothermal_vent/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
```

Comment:
I guess there's nothing wrong with this per se, but traditionally there's a generic processing SS for things which aren't objs. Don't personally care much.